File: CapaDatos/Models/UsuarioCostoMensualModel.cs
using System;
using System.Collections.Generic;

namespace CapaDatos.Models
{
    public class UsuarioCostoMensualModel
    {

        public long IdUsuarioCostoMensual { get; set; }
        public Nullable<long> IdUsuario { get; set; }
        public Nullable<int> Anio { get; set; }
        public Nullable<int> Mes { get; set; }
        public string NombreMes { get; set; }
        public Nullable<decimal> CostoMensual { get; set; }
        public long IdUCreo { get; set; }
        public long? IdULider { get; set; }
        public System.DateTime FechaCreo { get; set; }
        public Nullable<long> IdUModifico { get; set; }
        public Nullable<System.DateTime> FechaModifico { get; set; }

        public List<UsuarioCostoDistribucionModel> LstDistrbucion { get; set; }

        public string Nombre { get; set; }
        public string Clave { get; set; }
    }
    public class CostoAnualModel
    {
        public int Mes { get; set; }
        public string NombreMes { get; set; }
        public decimal TotalCosto { get; set; }
        public int TotalRecursos { get; set; }
    }
}

File: SISPRO/yitpro-react/src/model/UsuarioCostoMensual.model.ts
export interface ProyectoDistribucion {
    proyecto: string;
    porcentaje: number;
}

export interface UsuarioCosto {
    clave: string;
    nombre: string;
    nombreMes: string;
    anio: number;
    costoMensual: number;
    lstDistrbucion: ProyectoDistribucion[];
    idUsuario: number;
    idULider?: number | null;
    idUsuarioCosto: number;
    costoHora: number;
}

export interface UsuarioNodo extends UsuarioCosto {
    hijos?: UsuarioNodo[];
}

export interface UsuarioCostoDistribucion {
    idUsuarioCostoD: number;
    anio: number;
    mes: number;
    idUsuario: number;
    idProyecto: number;
    proyecto: string;
    porcentaje: number;
    idUCreo: number | null;
    fechaCreo: string | null;
    claveProy: string;
    nombreMes: string;
    clave: string;
    recurso: string;
    totalMes: number;
}
export interface CostoAnualModel {
    mes: number;
    nombreMes: string;
    totalCosto: number;
    totalRecursos: number;
}